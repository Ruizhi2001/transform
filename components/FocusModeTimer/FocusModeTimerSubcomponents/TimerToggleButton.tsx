import React from "react";
import { Pressable, View, StyleSheet, Text } from "react-native";
import { AntDesign } from "@expo/vector-icons";

type Props = {
  isTimerRunning: boolean;
  stopTimer: () => void;
  startTimer: () => void;
};

export const TimerToggleButton: React.FC<Props> = ({
  isTimerRunning,
  startTimer,
  stopTimer,
}) => {
  return (
    <Pressable
      onPress={isTimerRunning ? stopTimer : startTimer}
      style={styles.pressable}
    >
      <View>
        <Text style={styles.text}>
          {isTimerRunning ? "ðŸ˜¢ Give up ðŸ˜¢" : "âš¡TRANSFORM NOW!âš¡"}
        </Text>
      </View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  text: {
    fontSize: 30,
    fontWeight: "700",
  },
  pressable: {
    borderColor: "gold",
    borderWidth: 3,
    borderRadius: 10,
    alignItems: "center",
  },
});
