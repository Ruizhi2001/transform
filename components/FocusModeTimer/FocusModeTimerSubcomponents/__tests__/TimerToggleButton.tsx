import renderer from "react-test-renderer";
import { TimerToggleButton } from "../TimerToggleButton";

describe("TimerToggleButton", () => {
  it("should match the snapshot when timer is running", () => {
    const tree = renderer
      .create(
        <TimerToggleButton
          isTimerRunning={true}
          startTimer={jest.fn}
          stopTimer={jest.fn}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match the snapshot when timer is not running", () => {
    const tree = renderer
      .create(
        <TimerToggleButton
          isTimerRunning={false}
          startTimer={jest.fn}
          stopTimer={jest.fn}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
