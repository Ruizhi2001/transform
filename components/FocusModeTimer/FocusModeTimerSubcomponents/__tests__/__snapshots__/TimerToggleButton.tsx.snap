// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerToggleButton should match the snapshot when timer is not running 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "gold",
      "borderRadius": 10,
      "borderWidth": 3,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 30,
          "fontWeight": "700",
        }
      }
    >
      âš¡TRANSFORM NOW!âš¡
    </Text>
  </View>
</View>
`;

exports[`TimerToggleButton should match the snapshot when timer is running 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "gold",
      "borderRadius": 10,
      "borderWidth": 3,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 30,
          "fontWeight": "700",
        }
      }
    >
      ðŸ˜¢ Give up ðŸ˜¢
    </Text>
  </View>
</View>
`;
